### ISSUE TITLE
PSR1 and PSR2 standards are missing

### ISSUE THREAD

**Issue by glen-84** [PHPsrc/PHP-Tool-Integration#7]
Does this plug-in use an old version of PHP CodeSniffer?

**Comment by borgogelli** [comment]
Issue confirmed under Eclipse Neon

**Comment by binarious** [comment]
I got around that by forcing my locally installed CodeSniffer:

```
/Applications/Eclipse.app/Contents/Eclipse/plugins/org.phpsrc.eclipse.pti.tools.codesniffer_1.3.0.R20111119000000/php/tools/phpcs.php
```

Change include line to the following:

```
include_once '/Users/youruser/.composer/vendor/squizlabs/php_codesniffer/CodeSniffer/CLI.php';
```

After that I referenced PSR2 in the settings via this path:

```
/Users/youruser/.composer/vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/PSR2
```

**Comment by Kallys** [comment]
Definitely, this plugin uses an old version of PHP_CodeSniffer : with eclipse Neon 4.3 and PDT 4.3, I got PHP_CodeSniffer 1.3, released in 2011... ([latest release](https://github.com/squizlabs/PHP_CodeSniffer/releases) is 2.8.1, march 2017)

After trying many workarounds like installing PEAR manually, working aroung with "pti_phpcs.launch" file, etc. I was about to give up to get PSR2 support in eclipse, until I ended up on @binarious answer. His workaround actually still works! Thanks!

So here is another way (which I found less hacky) to update PHP_CodeSniffer support in eclipse with PTI (so it's working with PDT-ext):

1. Find where the PHP_CodeSniffer execution file ('phpcs.php') is.
It depends on your eclipse installation. You can use the "print PHP output to console" in "PHP CodeSniffer" properties in eclipse. Mine was here:
`/opt/.p2/pool/plugins/org.phpsrc.eclipse.pti.tools.codesniffer_1.3.0.R20111119000000/php/tools/phpcs.php`
2. Make a symbolic link in the `php` folder (parent of parent of 'phpcs.php') to your PHP_CodeSniffer installation path.
No need to edit the phpcs.php file since it is actually searching for a local installation of PHP_CodeSniffer inside the `php` folder.
`ln -s /opt/.composer/vendor/squizlabs/php_codesniffer/CodeSniffer /opt/.p2/pool/plugins/org.phpsrc.eclipse.pti.tools.codesniffer_1.3.0.R20111119000000/php/`
3. Manually add wanted standards from your PHP_CodeSniffer installation in eclipse preferences.
Press New, name it (eg: "PSR1"), set path (eg: `/opt/.composer/vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/PSR1`)
4. Enjoy!

Note: [Installation methods of PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer#installation). I used composer global installation, adapt your paths depending on your installation.
___
**Edit**: you could also consider using [this fork](https://github.com/Webaib/net.overscale.eclipse.pti.tool.codesniffer).
